module/modmanager/manager_test.go

```diff
@@ -1270,3 +1270,126 @@ func TestRTPPassthrough(t *testing.T) {
+	fakeRobot := rtestutils.MakeRobotForModuleLogging(t, parentAddr)
+	defer func() {
+		test.That(t, fakeRobot.Stop(), test.ShouldBeNil)
+	}()
+
+	greenLog(t, "test Helpers")
+	viamHomeTemp := t.TempDir()
+	mod := &module{
+		cfg: config.Module{
+			Name:     "test",
+			ExePath:  modPath,
+			Type:     config.ModuleTypeRegistry,
+			ModuleID: "rtppassthrough:camera",
+		},
+		dataDir: "module-data-dir",
+		logger:  logger,
+	}
+
+	err = mod.startProcess(ctx, parentAddr, nil, logger, viamHomeTemp)
+	test.That(t, err, test.ShouldBeNil)
+
+	err = mod.dial()
+	test.That(t, err, test.ShouldBeNil)
+
+	err = mod.checkReady(ctx, parentAddr, logger)
+	test.That(t, err, test.ShouldBeNil)
+
+	greenLog(t, "test AddModule")
+	mgr := NewManager(ctx, parentAddr, logger, modmanageroptions.Options{UntrustedEnv: false})
+	test.That(t, err, test.ShouldBeNil)
+
+	modCfg := config.Module{
+		Name:    "rtp-passthrough-module",
+		ExePath: modPath,
+	}
+	err = mgr.Add(ctx, modCfg)
+	test.That(t, err, test.ShouldBeNil)
+
+	reg, ok := resource.LookupRegistration(camera.API, model)
+	test.That(t, ok, test.ShouldBeTrue)
+	test.That(t, reg.Constructor, test.ShouldNotBeNil)
+
+	greenLog(t, "add 10 cameras")
+	cams := []resource.Resource{}
+	for _, conf := range confs {
+		cam, err := mgr.AddResource(ctx, conf, nil)
+		test.That(t, err, test.ShouldBeNil)
+		cams = append(cams, cam)
+	}
+
+	sources := []rtppassthrough.Source{}
+	for _, cam := range cams {
+		source, ok := cam.(rtppassthrough.Source)
+		test.That(t, ok, test.ShouldBeTrue)
+		sources = append(sources, source)
+	}
+
+	first9Sources := sources[1:]
+
+	test.That(t, len(first9Sources), test.ShouldEqual, 9)
+
+	greenLog(t, "the first 9's SubscribeRTP calls succeed")
+	subscribeRTPTimeout := time.Second * 5
+	for _, source := range first9Sources {
+		calledCtx, calledFn := context.WithCancel(context.Background())
+		// SubscribeRTP succeeds
+		subscribeRTPCtx, subscribeRTPcancelFn := context.WithTimeout(context.Background(), subscribeRTPTimeout)
+		sub, err := source.SubscribeRTP(subscribeRTPCtx, 512, func(pkts []*rtp.Packet) {
+			test.That(t, len(pkts), test.ShouldBeGreaterThan, 0)
+			calledFn()
+		})
+		subscribeRTPcancelFn()
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, sub.ID, test.ShouldNotResemble, rtppassthrough.NilSubscription)
+		test.That(t, sub.Terminated.Err(), test.ShouldBeNil)
+		<-calledCtx.Done()
+	}
+
+	greenLog(t, "the 10th returns an error")
+	subscribeRTPCtx, subscribeRTPcancelFn := context.WithTimeout(context.Background(), subscribeRTPTimeout)
+	sub, err := sources[0].SubscribeRTP(subscribeRTPCtx, 512, func(pkts []*rtp.Packet) {
+		t.Log("should not happen")
+		t.FailNow()
+	})
+	subscribeRTPcancelFn()
+	test.That(t, err, test.ShouldBeError)
+	test.That(t, err.Error(), test.ShouldContainSubstring, "only 9 WebRTC tracks are supported per peer connection")
+	test.That(t, sub, test.ShouldResemble, rtppassthrough.NilSubscription)
+
+	greenLog(t, "cleanup")
+	err = mgr.Close(ctx)
```

Comment:
Similar to the below comment, I'd expect this in a defer.

Reply:
done